src/node_env_var.cc

```diff
@@ -316,6 +316,24 @@ Maybe<bool> KVStore::AssignFromObject(Local<Context> context,
   return Just(true);
 }
 
+// TODO(bnoordhuis) Not super efficient but called infrequently. Not worth
+// the trouble yet of specializing for RealEnvStore and MapKVStore.
+void KVStore::AssignToObject(v8::Isolate* isolate,
+                             v8::Local<v8::Context> context,
+                             v8::Local<v8::Object> object) {
+  HandleScope scope(isolate);
+  Local<Array> keys = Enumerate(isolate);
+  uint32_t keys_length = keys->Length();
+  for (uint32_t i = 0; i < keys_length; i++) {
+    Local<Value> key;
+    Local<String> value;
+    if (!keys->Get(context, i).ToLocal(&key)) continue;
+    if (!key->IsString()) continue;
+    if (!Get(isolate, key.As<String>()).ToLocal(&value)) continue;
+    USE(object->Set(context, key, value));
+  }
```

Comment:
Wahat about exception handling?

```suggestion
Maybe<bool> KVStore::AssignToObject(v8::Isolate* isolate,
                                    v8::Local<v8::Context> context,
                                    v8::Local<v8::Object> object) {
  HandleScope scope(isolate);
  Local<Array> keys = Enumerate(isolate);
  uint32_t keys_length = keys->Length();
  for (uint32_t i = 0; i < keys_length; i++) {
    Local<Value> key;
    Local<String> value;
    if (!keys->Get(context, i).ToLocal(&key)) return Nothing<bool>();
    if (!key->IsString()) continue;
    if (!Get(isolate, key.As<String>()).ToLocal(&value)) return Nothing<bool>();
    if (object->Set(context, key, value).IsNothing()) return Nothing<bool>();
  }
  return Just(true);
```

At least the last `Set()` can fail for `object` with a non-null prototype, but even if that object were to be guaranteed to have a null prototype, I'd like it if we could stick to proper `Maybe` propagation to avoid brittle code that then gets copied to other parts of the source

Reply:
Good point, done. I used `ok = ok && ...` to keep line length in check.